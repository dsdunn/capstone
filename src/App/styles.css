.test-img {
  opacity: 0.5; }

* {
  box-sizing: border-box;
  font-family: 'Open Sans', sans-serif; }

.App {
  height: 100vh;
  position: relative; }

form {
  max-width: 720px; }

a:hover {
  cursor: pointer; }

@media screen and (max-width: 720px) {
  form {
    width: 98%;
    margin: auto; } }
