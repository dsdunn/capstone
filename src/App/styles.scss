.test-img {
  opacity: 0.5;
}

.App {
  height: 100vh;
  background-image: url('https://images.pexels.com/photos/157519/pexels-photo-157519.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260');
  background-size: cover;
}