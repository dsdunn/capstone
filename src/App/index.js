import React, { Component } from 'react';
import { Route, withRouter } from 'react-router-dom';
import './normalize.css';
import './styles.css';
import Header from '../Header'
import UserProfile from '../UserProfile'
import Login from '../Login';
import Signup from '../Signup';
import EditProfile from '../EditProfile';
import LandingPage from '../LandingPage';
import CollectionBig from '../CollectionBig';
import AddCollection from '../AddCollection';
import CollectionsContainer from '../CollectionsContainer';
import EditCollection from '../EditCollection';
import { connect } from 'react-redux';
import { signIn, setProfile } from '../actions';
import { getUserInfo } from '../services/fetch';
import { firebase } from '../firebase';


export class App extends Component {
  componentDidMount() {
    firebase.auth.onAuthStateChanged(authUser => {
      if (authUser) {
        getUserInfo(authUser.uid)
          .then(user => {
            this.props.signIn(user)
            this.props.setProfile(user)
          })
      }
    })
  }

  render() {
    return (
      <div className="App">
        <Header />
        <Route path={'/user/addcollection'} component={AddCollection} />
        <Route exact path={'/'} component={LandingPage} />
        <Route exact path={'/login'} render={() => (<div><LandingPage/><Login/></div>)} />
        <Route exact path={'/user/login'} render={() => (<div><UserProfile/><Login/></div>)} />
        <Route exact path={'/home/login'} render={() => (<div><CollectionsContainer/><Login/></div>)} />
        <Route path={'/signup'} render={() => (<div><LandingPage/><Signup/></div>)} />
        <Route path={'/user/signup'} render={Signup} />
        <Route path={'/home/signup'} component={Signup} />
        <Route path={'/user'} component={UserProfile} />
        <Route path={'/user/editprofile'} component={EditProfile} />
        <Route path={'/home/editprofile'} component={EditProfile} />
        <Route path={'/collection'} component={CollectionBig}/>
        <Route path={'/home'} component={CollectionsContainer}/>
        <Route path={'/collection/editcollection'} component={EditCollection}/>
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  signIn: (userId) => dispatch(signIn(userId)),
  setProfile: (user) => dispatch(setProfile(user))
})

export default withRouter(connect(null, mapDispatchToProps)(App));
