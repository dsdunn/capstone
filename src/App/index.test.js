import React from 'react';
import App from './';
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper = shallow(<App/>);
  it('should match the snapShot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
