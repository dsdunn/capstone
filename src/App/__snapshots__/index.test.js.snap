// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapShot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <withRouter(Connect(Header)) />,
        <Route
          component={[Function]}
          path="/user/addcollection"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/"
/>,
        <Route
          exact={true}
          path="/login"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/user/login"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/home/login"
          render={[Function]}
/>,
        <Route
          path="/signup"
          render={[Function]}
/>,
        <Route
          path="/user/signup"
          render={[Function]}
/>,
        <Route
          component={[Function]}
          path="/home/signup"
/>,
        <Route
          component={[Function]}
          path="/user"
/>,
        <Route
          component={[Function]}
          path="/user/editprofile"
/>,
        <Route
          component={[Function]}
          path="/home/editprofile"
/>,
        <Route
          component={[Function]}
          path="/collection"
/>,
        <Route
          component={[Function]}
          path="/home"
/>,
        <Route
          component={[Function]}
          path="/collection/editcollection"
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/user/addcollection",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/login",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/user/login",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/home/login",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "path": "/signup",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "path": "/user/signup",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/home/signup",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/user",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/user/editprofile",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/home/editprofile",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/collection",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/home",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/collection/editcollection",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter(Connect(Header)) />,
          <Route
            component={[Function]}
            path="/user/addcollection"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/"
/>,
          <Route
            exact={true}
            path="/login"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/user/login"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/home/login"
            render={[Function]}
/>,
          <Route
            path="/signup"
            render={[Function]}
/>,
          <Route
            path="/user/signup"
            render={[Function]}
/>,
          <Route
            component={[Function]}
            path="/home/signup"
/>,
          <Route
            component={[Function]}
            path="/user"
/>,
          <Route
            component={[Function]}
            path="/user/editprofile"
/>,
          <Route
            component={[Function]}
            path="/home/editprofile"
/>,
          <Route
            component={[Function]}
            path="/collection"
/>,
          <Route
            component={[Function]}
            path="/home"
/>,
          <Route
            component={[Function]}
            path="/collection/editcollection"
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/user/addcollection",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/login",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/user/login",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/home/login",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/signup",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/user/signup",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/home/signup",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/user",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/user/editprofile",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/home/editprofile",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/collection",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/home",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/collection/editcollection",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
